torchtext/experimental/transforms.py

```diff
@@ -340,3 +341,147 @@ def forward(self, tokens: List[str]) -> Tensor:
         """
 
         return self.vector.lookup_vectors(tokens)
+
+
+class MaskTransform(nn.Module):
+    """
+    The transform chooses mask_prob% (example 15%) of the token positions at random for
+    prediction.
+
+    If the i-th token is chosen, we replace the i-th token with
+    (1) the [MASK] token 80% of the time
+    (2) a random token 10% of the time
+    (3) the unchanged i-th token 10% of the time.
+
+    NOTE: this transform is not torchscriptible.
+
+    Args:
+        vocab_len (int): the length of the vocabulary, including special tokens such as [BOS], [PAD], [MASK]
+        mask_idx (int): index assigned to mask token in vocabulary
+        bos_idx (int): index assigned to beginning-of-sequence token in vocabulary
+        pad_idx (int): index assigned to padding token in vocabulary
+        mask_bos (bool): indicate whether beginning-of-sequence tokens are eligible for masking (default: False)
+        mask_prob (float): probability that a token is chosen for replacement (default: 0.15)
+
+    Example:
+        >>> import torch
+        >>> from torchtext.experimental.transforms import MaskTransform
+        >>> sample_tokens = [
+                ["[BOS]", "a", "b", "c", "d"],
+                ["[BOS]", "a", "b", "[PAD]", "[PAD]"]
+            ]
+        >>> sample_token_ids = torch.tensor([
+                [6, 0, 1, 2, 3], [6, 0, 1, 4, 4]
+            ])
+        >>> mask_transform = MaskTransform(
+                vocab_len = 7,
+                mask_idx = 4,
+                bos_idx = 6,
+                pad_idx = 5,
+                mask_bos = False,
+                mask_prob = 0.15
+            )
+        >>> masked_tokens, target_tokens, mask = mask_transform(sample_token_ids)
+    """
+
+    # maks_mask_prob is prob. of replacing a token with [MASK] (ex. 80%)
+    mask_mask_prob = 0.8
+
+    # rand_mask_thresh is prob. of replacing a token with a random token. (ex.10%)
+    rand_mask_prob = 0.1
+
+    def __init__(
+        self,
+        vocab_len: int,
+        mask_idx: int,
+        bos_idx: int,
+        pad_idx: int,
+        mask_bos: bool = False,
+        mask_prob: float = 0.15,
+    ):
+        super().__init__()
+        self.vocab_len: int = vocab_len
+        self.mask_idx: int = mask_idx
+        self.bos_idx: int = bos_idx
+        self.pad_idx: int = pad_idx
+        self.mask_prob: float = mask_prob
+        self.mask_bos: bool = mask_bos
+
+    def forward(self, tokens: torch.Tensor) -> Tuple[torch.Tensor, torch.Tensor, torch.Tensor]:
+        """
+        Applies mask to input tokens.
+
+        Inputs:
+            tokens: Tensor with token ids of shape (batch_size x seq_len)
```

Comment:
Let's also mention that that tokens are expected to contain any special tokens like [BOS] and [PAD] tokens ids.